### PR TITLE
Adds symbolic op for logsumexp

### PR DESCRIPTION
**Pull request by iiSeymour**

### CODE REVIEW

**Comment by lara-hdr** [comment]
The caffe2 test is failing. 
I looked in https://caffe2.ai/docs/operators-catalogue.html and it seems like ReduceLogSumExp op does not exist in caffe2.
Try adding an ONNXRuntime test in test/onnx/test_pytorch_onnx_onnxruntime.py to verify the behavior?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -129,6 +129,13 @@ def forward(self, x):
         x = torch.randn(1, 2, 3, 4, requires_grad=True)
         self.run_test(MyModel(), x)
 
+    def test_reduce_log_sum_exp(self):
+        class ReduceLogSumExpModel(torch.nn.Module):
+            def forward(self, input):
+                return torch.logsumexp(input, dim=(0, 1))
+
+        x = torch.randn(4, 4, requires_grad=True)
+        self.run_model_test(ReduceLogSumExpModel(), x)
```
Comment (lara-hdr) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: self.run_test instead of self.run_model_test.

you can also test locally by running the command "python test/onnx/test_pytorch_onnx_onnxruntime.py TestONNXRuntime.test_reduce_log_sum_exp" :)

Comment (iiSeymour) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: D'oh, c&p fail. thanks!

**Comment by iiSeymour** [comment]
The `caffe2-py2-devtoolset7-rocmrpm-centos7.5` failure appears to be unrelated to the change.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -129,6 +129,13 @@ def forward(self, x):
         x = torch.randn(1, 2, 3, 4, requires_grad=True)
         self.run_test(MyModel(), x)
 
+    def test_reduce_log_sum_exp(self):
+        class ReduceLogSumExpModel(torch.nn.Module):
+            def forward(self, input):
+                return torch.logsumexp(input, dim=(0, 1))
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_onnxruntime.py]: also test single dimension?

something like torch.logsumexp(input, dim=0) + torch.logsumexp(input, dim=(0, 1)

**Review by houseroad — COMMENTED** [review]
Looks good. If the test can cover the single dimension case, it will be better.

Thanks!

**Comment by iiSeymour** [comment]
Single dim test added & tests passing :heavy_check_mark:

**Review by houseroad — APPROVED** [review]
LGTM, thanks.
